projects/opendr_ws/src/perception/scripts/object_detection_2d_gem.py

```diff
@@ -175,26 +223,66 @@ def callback(self, msg_rgb, msg_ir):
             # We get can the data back using self.bridge.from_ros_bounding_box_list(ros_detection)
             # e.g., opendr_detection = self.bridge.from_ros_bounding_box_list(ros_detection)
 
-        if self.rgb_publisher is not None:
-            plot_rgb = draw(image_rgb, boxes, w_sensor1, mean_fps)
-            message = self.bridge.to_ros_image(Image(np.uint8(plot_rgb)))
-            self.rgb_publisher.publish(message)
+        if self.color_publisher is not None:
+            plot_color = draw_bounding_boxes(image_color, boxes, class_names=self.classes)
+            message = self.bridge.to_ros_image(Image(np.uint8(plot_color)))
+            self.color_publisher.publish(message)
         if self.ir_publisher is not None:
-            plot_ir = draw(image_ir, boxes, w_sensor1, mean_fps)
+            plot_ir = draw_bounding_boxes(image_ir, boxes, class_names=self.classes)
             message = self.bridge.to_ros_image(Image(np.uint8(plot_ir)))
             self.ir_publisher.publish(message)
 
 
-if __name__ == '__main__':
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--input_color_image_topic", help="Topic name for input color image", type=str, default="/camera/color/image_raw"
+    )
+    parser.add_argument(
+        "--output_color_image_topic",
+        help="Topic name for output annotated color image",
+        type=str,
+        default="/opendr/color_objects_annotated",
+    )
```

Comment:
Similar to the constructor arguments, "color" should be "rgb" for consistency with other nodes. 

Moreover, other object detection nodes that only output annotated RGB images use the  "image_objects_annotated" default value for the output topic. This node has both infrared and RGB annotated images, so for consistency i would suggest that it could become "rgb_image_objects_annotated" and similarly, the infrared output topic should be "infra_image_objects_annotated". Whatever default value you choose should be also changed in the ctor arguments, docstrings, etc.